pkg/kube/inject/openshift.go

```diff
@@ -0,0 +1,113 @@
+
+// Functions below were copied from
+// https://github.com/openshift/apiserver-library-go/blob/c22aa58bb57416b9f9f190957d07c9e7669c26df/pkg/securitycontextconstraints/sccmatching/matcher.go
+// These functions are not exported, and, if they were, when imported bring k8s.io/kubernetes as dependency, which is problematic
+// License is Apache 2.0: https://github.com/openshift/apiserver-library-go/blob/c22aa58bb57416b9f9f190957d07c9e7669c26df/LICENSE
+
+package inject
+
+import (
+	"fmt"
+	"strings"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	"github.com/openshift/library-go/pkg/security/uid"
+	corev1 "k8s.io/api/core/v1"
+
+	"istio.io/istio/pkg/log"
+)
+
+// getPreallocatedUIDRange retrieves the annotated value from the namespace, splits it to make
+// the min/max and formats the data into the necessary types for the strategy options.
+func getPreallocatedUIDRange(ns *corev1.Namespace) (*int64, *int64, error) {
+	annotationVal, ok := ns.Annotations[securityv1.UIDRangeAnnotation]
+	if !ok {
+		return nil, nil, fmt.Errorf("unable to find annotation %s", securityv1.UIDRangeAnnotation)
+	}
+	if len(annotationVal) == 0 {
+		return nil, nil, fmt.Errorf("found annotation %s but it was empty", securityv1.UIDRangeAnnotation)
+	}
+	uidBlock, err := uid.ParseBlock(annotationVal)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	min := int64(uidBlock.Start)
+	max := int64(uidBlock.End)
+	log.Debugf("got preallocated values for min: %d, max: %d for uid range in namespace %s", min, max, ns.Name)
+	return &min, &max, nil
+}
+
+// getPreallocatedSupplementalGroups gets the annotated value from the namespace.
+func getPreallocatedSupplementalGroups(ns *corev1.Namespace) ([]securityv1.IDRange, error) {
+	groups, err := getSupplementalGroupsAnnotation(ns)
+	if err != nil {
+		return nil, err
+	}
+	log.Debugf("got preallocated value for groups: %s in namespace %s", groups, ns.Name)
+
+	blocks, err := parseSupplementalGroupAnnotation(groups)
+	if err != nil {
+		return nil, err
+	}
+
+	idRanges := []securityv1.IDRange{}
+	for _, block := range blocks {
+		rng := securityv1.IDRange{
+			Min: int64(block.Start),
+			Max: int64(block.End),
+		}
+		idRanges = append(idRanges, rng)
+	}
+	return idRanges, nil
+}
+
+// getSupplementalGroupsAnnotation provides a backwards compatible way to get supplemental groups
+// annotations from a namespace by looking for SupplementalGroupsAnnotation and falling back to
+// UIDRangeAnnotation if it is not found.
+func getSupplementalGroupsAnnotation(ns *corev1.Namespace) (string, error) {
+	groups, ok := ns.Annotations[securityv1.SupplementalGroupsAnnotation]
+	if !ok {
+		log.Debugf("unable to find supplemental group annotation %s falling back to %s", securityv1.SupplementalGroupsAnnotation, securityv1.UIDRangeAnnotation)
+
+		groups, ok = ns.Annotations[securityv1.UIDRangeAnnotation]
+		if !ok {
+			return "", fmt.Errorf("unable to find supplemental group or uid annotation for namespace %s", ns.Name)
```

Comment:
similar to the log statement above you should print out `securityv1.UIDRangeAnnotation`

Reply:
This is a copy of upstream (see the header in the file). I'd keep the differences minimum.